src/libexpr/eval-settings.hh

```diff
@@ -50,6 +51,17 @@ struct EvalSettings : Config
 
     LookupPathHooks lookupPathHooks;
 
+    std::vector<PrimOp> extraPrimOps;
+
+    /**
+     * Register a primop to be added when an EvalState is created from these settings.
+     */
+    void addPrimOp(PrimOp && primOp);
+    /**
+     * Register a primop to be added when an EvalState is created from these settings.
+     */
+    void addPrimOp(const PrimOp & primOp);
+
```

Comment:
Is this really worth it, as opposed to just calling `emplace_back` on the underlying vector? IMO it is confusing to write the function when there is nothing we're trying to encapsulate --- the caller is free to manipulate `extraPrimOps` however they want before using the settings to make an `EvalState`.